src/routes/stops/[stopID]/schedule/+page.svelte

```diff
@@ -0,0 +1,185 @@
+		} catch (error) {
+			console.error('Error fetching schedules:', error);
+		} finally {
+			loading = false;
+		}
+	}
+
+	function handleScheduleForStopResponse(scheduleForStop) {
+		schedulesMap.clear();
+		routeReference.clear();
+
+		if (!scheduleForStop.entry.stopRouteSchedules.length) {
+			emptySchedules = true;
+			schedules = [];
+			expandedItems = [];
+			return;
+		}
+
+		setStopDetails(scheduleForStop.references.stops[0]);
+		mapRoutes(scheduleForStop.references.routes);
+		processRouteSchedules(scheduleForStop.entry.stopRouteSchedules);
+
+		schedules = Array.from(schedulesMap.values());
+		expandedItems = schedules.map(() => false);
+	}
+
+	function mapRoutes(routes) {
+		for (let route of routes) {
+			routeReference.set(route.id, route);
+		}
+	}
+
+	function setStopDetails(stop) {
+		stopName = stop.name;
+		stopId = stop.id;
+		stopDirection = stop.direction;
+	}
+
+	function processRouteSchedules(routeSchedules) {
+		for (let routeSchedule of routeSchedules) {
+			let routeId = routeSchedule.routeId;
+			let stopRouteDirectionSchedules = routeSchedule.stopRouteDirectionSchedules;
+
+			stopRouteDirectionSchedules.forEach((directionSchedule) => {
+				const stopTimesGroupedByHour = groupStopTimesByHour(directionSchedule.scheduleStopTimes);
+				const routeName = getRouteName(routeId, directionSchedule.tripHeadsign);
+
+				schedulesMap.set(routeName, {
+					tripHeadsign: routeName,
+					stopTimes: stopTimesGroupedByHour
+				});
+			});
+		}
+	}
+
+	function getRouteName(routeId, tripHeadsign) {
+		const route = routeReference.get(routeId);
+		return `${route.shortName ?? route.longName} - ${tripHeadsign}`;
+	}
+
+	function groupStopTimesByHour(stopTimes) {
+		const grouped = {};
+		for (let stopTime of stopTimes) {
+			const date = new Date(stopTime.arrivalTime);
+			const hour = date.getHours();
+			if (!grouped[hour]) grouped[hour] = [];
+			grouped[hour].push({
+				arrivalTime: formatTime(stopTime.arrivalTime)
+			});
+		}
+		return grouped;
+	}
+
+	function toggleAll(expand) {
+		expandedItems = schedules.map(() => expand);
+	}
+
+	function toggleAccordion(index) {
+		expandedItems = expandedItems.map((item, i) => (i === index ? !item : item));
+	}
+
+	onMount(() => {
+		const formattedDate = currentDate.toISOString().split('T')[0];
+		fetchScheduleForStop(stopId, formattedDate);
+	});
+</script>
+
+{#if loading || $isLoading}
+	<LoadingSpinner />
+{:else}
+	<div class="mx-auto max-w-7xl overflow-y-auto p-5" style="max-height: calc(100vh - 100px);">
+		<StopDetailsHeader {stopName} {stopId} {stopDirection} />
+
+		<div class="flex flex-col gap-6 md:flex-row">
+			<div class="md:w-1/3">
+				<h2 class="mb-4 text-xl font-semibold text-gray-800">Select Date</h2>
+				<div class="rounded-lg border border-gray-300 bg-white p-4 shadow">
+					<Datepicker inline bind:value={selectedDate} />
+					<p class="mt-4 text-sm text-gray-600">
+						Selected date: {selectedDate
```

Comment:
`t()` — here and elsewhere in the file

Reply:
Yes, if everything works out in the end, I was going to add the localization.